### PR TITLE
fix(sdk): making cron schedule more cloud agnostic

### LINKED ISSUE #2849: Schedule fails to compile when using `cron: "* * * * *"`
### I tried this:

```sh
wing compile cron.w -t sim                     
```

```ts
// cron.w
bring cloud;
bring redis;

let cronSchedule = new cloud.Schedule(cloud.ScheduleProps{
  cron: "* * * * *"
}) as "Cron Schedule";

cronSchedule.onTick(inflight () => {
  log("Cron schedule ticked!");
});
```

### This happened:

```
Error: ERROR: cannot use * in both the Day-of-month and Day-of-week fields. If you use it in one, you must use ? in the other

demo/target/index.wsim.294604.tmp/.wing/preflight.js:513
       );
       (rateSchedule.onTick(new $Closure8(this,"$Closure8")));
>>     const cronSchedule = this.node.root.newAbstract("@winglang/sdk.cloud.Schedule",this,"Cron Schedule",{
       "cron": "* * * * *",}
       );
```

### I expected this:

It should compile and the scheduler should tick every minute.

### Is there a workaround?

I don't know.

### Component

Compiler, SDK

### Wing Version

0.19.4

### Wing Console Version

_No response_

### Node.js Version

18.16.0

### Platform(s)

MacOS

### Anything else?

_No response_

### Community Notes

* Please vote by adding a 👍 reaction to the issue to help us prioritize.
* If you are interested to work on this issue, please leave a comment.

### PR DESCRIPTION
**Pull request by marciocadev**
Some characters like `?` are configurations used by AWS, but to make the schedule more cloud-agnostic, I made some changes to bring it closer to the Unix implementation

Now a cron like `* * * * *` will not throw an exception.

Closes #2849 

## Checklist

- [x] Title matches [Winglang's style guide](https://www.winglang.io/contributing/start-here/pull_requests#how-are-pull-request-titles-formatted)
- [x] Description explains motivation and solution
- [x] Tests added (always)
- [ ] Docs updated (only required for features)
- [ ] Added `pr/e2e-full` label if this feature requires end-to-end testing

*By submitting this pull request, I confirm that my contribution is made under the terms of the [Wing Cloud Contribution License](https://github.com/winglang/wing/blob/main/CONTRIBUTION_LICENSE.md)*.

### CODE REVIEW

File: libs/wingsdk/src/shared-aws/schedule.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+   * The implementation of cron on AWS does not allow [day of month] and [day of week]
+   * to have the character '*' at the same time.
+   * Therefore, [day of week] will be replaced by '?'.
+   */
+  if (cron && dayOfMonth == "*" && dayOfWeek == "*") {
+    dayOfWeek = "?";
+  }
+
+  if (cron && dayOfMonth !== "*" && dayOfWeek !== "*") {
+    throw new Error("Cannot supply both 'day' and 'weekDay', use at most one");
```
Comment (Chriscbr) [inline libs/wingsdk/src/shared-aws/schedule.ts:21]:
```suggestion
    throw new Error("Cannot restrict both 'day-of-month' and 'day-of-week' in a cron expression, at least one must be '*'");
```

**Review by Chriscbr — APPROVED** [review]
Nice work, thanks @marciocadev !

**Comment by staycoolcall911** [comment]
This PR was reverted due to a failing sdk spec test: https://github.com/winglang/wing/actions/runs/8304298940/job/22729804328#step:12:1

**Comment by staycoolcall911** [comment]
Sorry @marciocadev - when youu get the chance can you please take a look? Happy to help and run the sdk spec tests on the fix PR

**Comment by marciocadev** [comment]
@staycoolcall911 

The problem was that there was an error test that used to fail because the cron was only working properly for AWS. Now, I've changed it to accept the format used by UNIX, and I fixed the issue that was previously occurring. Essentially, I had to remove one of the validations because it no longer applies.

**Pull request merged**